test/integration/router_test.go

```diff
@@ -464,6 +464,10 @@ func TestRouterPathSpecificity(t *testing.T) {
 	if _, err := getRoute(routeAddress, "www.example.com", "http", nil, ""); err != ErrUnavailable {
 		t.Fatalf("unexpected response: %q", err)
 	}
+	//ensure you can curl path with port in Host header
+	if err := waitForRoute(routeTestAddress, "www.example.com:80", "http", nil, tr.HelloPodPath); err != nil {
```

Comment:
The second argument here gets put into the `Request.Host` field, which is what determines the `Host` header.